compiler/parse.py

```diff
@@ -6,23 +6,48 @@
 import subprocess
 import sys
 
+from ast_util import *
 from util import *
+from definitions.ast_node import *
 
 sys.path.append(os.path.join(config.PASH_TOP, "compiler/parser/ceda"))
+
+from ast2shell import *
 from parse_to_ast2 import parse_to_ast
-from json_to_shell2 import json_to_shell_string
+from json_to_shell2 import json_to_shell_string, json_string_to_shell_string
 
 ## Parses straight a shell script to an AST
 ## through python without calling it as an executable
-def parse_shell_to_ast(input_script_path):
-    new_asts = parse_to_ast(input_script_path)
-    return new_asts
-
-## TODO: Avoid going through JSON for the unparsing.
-## Parser straight from JSON to a shell string without calling an executable 
-def from_ir_to_shell(ir_filename):
-    preprocessed_script = json_to_shell_string(ir_filename)
-    return preprocessed_script
+def parse_shell_to_asts(input_script_path):
+    new_ast_objects = parse_to_ast(input_script_path)
+    return new_ast_objects
+
+def from_ast_objects_to_shell(asts):
+    shell_list = []
+    for ast in asts:
+        if(isinstance(ast, UnparsedScript)):
+            # log("Unparsed script is going back out")
+            # log(ast.text)
+            # exit(1)
+            shell_list.append(ast.text)
+        else:
+            ## We are working with two different abstractions for ASTs, one is the class and the other
+            ## is its JSON object form. Due to Python's _disgusting_ lack of types (and our bad code)
```

Comment:
😂

Reply:
this code = 💩 

I will open an issue to properly address this when I have some time :)